src/ModelContextProtocol/Logging/LoggerExtensions.cs

```diff
@@ -0,0 +1,35 @@
+using Microsoft.Extensions.Logging;
+
+namespace ModelContextProtocol.Logging;
+
+/// <summary>
+/// Extensions methods for ILogger instances used in MCP protocol.
+/// </summary>
+public static partial class LoggerExtensions
+{
+    /// <summary>
+    /// Logs the byte representation of a message in UTF-8 encoding.
+    /// </summary>
+    /// <param name="logger">The logger to use.</param>
+    /// <param name="endpointName">The name of the endpoint.</param>
+    /// <param name="byteRepresentation">The byte representation as a hex string.</param>
+    [LoggerMessage(EventId = 39000, Level = LogLevel.Trace, Message = "Transport {EndpointName}: Message bytes (UTF-8): {ByteRepresentation}")]
+    public static partial void TransportMessageBytes(this ILogger logger, string endpointName, string byteRepresentation);
```

Comment:
For now, let's move this to the Log.cs file where all the other log messages are. There's an open issue to refactor that, but I'd prefer until we do that to avoid the proliferation.